drivers/sensor/iis2mdc/iis2mdc.c

```diff
@@ -367,6 +322,80 @@ static int iis2mdc_init(const struct device *dev)
 	return 0;
 }
 
-DEVICE_DT_INST_DEFINE(0, iis2mdc_init, device_pm_control_nop,
-		     &iis2mdc_data, &iis2mdc_dev_config, POST_KERNEL,
-		     CONFIG_SENSOR_INIT_PRIORITY, &iis2mdc_driver_api);
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "IIS2MDC driver enabled without any devices"
+#endif
+
+/*
+ * Device creation macro, shared by IIS2MDC_DEFINE_SPI() and
+ * IIS2MDC_DEFINE_I2C().
+ */
+
+#define IIS2MDC_DEVICE_INIT(inst)					\
+	DEVICE_DT_INST_DEFINE(inst,					\
+			    iis2mdc_init,				\
+			    device_pm_control_nop,			\
+			    &iis2mdc_data_##inst,			\
+			    &iis2mdc_config_##inst,			\
+			    POST_KERNEL,				\
+			    CONFIG_SENSOR_INIT_PRIORITY,		\
+			    &iis2mdc_driver_api);
+
+/*
+ * Instantiation macros used when a device is on a SPI bus.
+ */
+
+#ifdef CONFIG_IIS2MDC_TRIGGER
+#define IIS2MDC_CFG_IRQ(inst) \
+		.drdy_port = DT_INST_GPIO_LABEL(inst, drdy_gpios),	\
+		.drdy_pin = DT_INST_GPIO_PIN(inst, drdy_gpios),		\
+		.drdy_flags = DT_INST_GPIO_FLAGS(inst, drdy_gpios),
```

Comment:
This is still using label properties. You can get the device directly and avoid other boilerplate by using `GPIO_DT_SPEC_GET(DT_DRV_INST(inst), drdy_gpios)`. See the mesh_demo sample for examples.

Reply:
> This is still using label properties. You can get the device directly and avoid other boilerplate by using `GPIO_DT_SPEC_GET(DT_DRV_INST(inst), drdy_gpios)`. See the mesh_demo sample for examples.

@mbolivar-nordic 
Seems ok, but I need to rebase the PR to latest master branch, as here GPIO_DT_SPEC_GET is not yet available.